src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -247,6 +247,10 @@ private static Assembly ResolveForKnownRequestor(string requestedAssemblyName, A
 
                     string directoryPath = Path.GetDirectoryName(requestingAssemblyFullPath);
                     string assemblyFullPath = Path.Combine(directoryPath, assemblyIdentity.Name + ".dll");
+                    if (!File.Exists(assemblyFullPath))
+                    {
+                        return null;
+                    }
 
```

Comment:
1. This is my theory: I think recently CLR assembly resolution seems to have undergone a change in that unhandled exceptions from custom assembly resolvers are not masked and go unhandled. When we moved to a new baseline drop with .NET 4.6 recently, we started seeing exceptions in custom assembly resolver in our Tao StartTargetAction (see our TFS changeset 1394695 and 1408896). The prior code was obviously wrong and would have thrown for both cases that were fixed, but it seemed this exception never showed up, and a subsequent resolver resolved the assembly. My _guess_ is before v4.6, CLR was catching the exceptions from custom assembly resolvers while trying to load these assemblies and moving ahead with next resolver to attempt resolution, but now it doesn't (at least not for Load in no context). Someone from CLR team should confirm though.

2. IDE scenario: In the IDE we never seemed to be getting a callback for the satellite assembly. I debugged through the CLR assembly resolution and it seems to be taking a different code path for IDE and command line scenarios (IDE: _http://referencesource.microsoft.com/#mscorlib/system/reflection/assembly.cs,3012_ while in command line case: _http://referencesource.microsoft.com/#mscorlib/system/reflection/assembly.cs,3028_ , the later one seems to call custom resolvers and former bails out without event doing so). It seems to be taking different paths on Assembly.CodeBase value, which I am not sure makes any sense for Load in no context. I am not going to dig further into this, but my best guess is Load in no context comes with its own adventures :)